en/ros2/px4_ros2_navigation_interface.md

```diff
@@ -0,0 +1,303 @@
+3. Pass the struct to the `GlobalPositionMeasurementInterface` [update()](https://auterion.github.io/px4-ros2-interface-lib/classpx4__ros2_1_1GlobalPositionMeasurementInterface.html#a1a183b595ef7f6a22f3a83ba543fe86d) method.
+
+The `GlobalPositionMeasurement` struct is defined as follows:
+
+```cpp
+struct GlobalPositionMeasurement
+{
+   rclcpp::Time timestamp_sample {};
+
+   std::optional<Eigen::Vector2d> lat_lon {std::nullopt};
+   std::optional<float> horizontal_variance {std::nullopt};
+
+   std::optional<float> altitude_msl {std::nullopt};
+   std::optional<float> vertical_variance {std::nullopt};
+};
+```
+
+The `update()` method of the global interface expects the following conditions to hold for `GlobalPositionMeasurement`:
+
+- The sample `timestamp_sample` is defined.
+- Values do not have a NAN.
+- If a measurement value is provided, its associated variance value is well defined (e.g. if `lat_lon` is defined, then `horizontal_variance` must be defined).
+
+The following code snippet is an example of a ROS 2 node which uses the global navigation interface to send a measurement with latitude, longitude and altitude to PX4:
+
+```cpp
+class MyGlobalMeasurementUpdateNode : public rclcpp::Node
+{
+public:
+   MyGlobalMeasurementUpdateNode()
+   : Node("my_node_name")
+   {
+      // Initialize global interface [1]
+      _global_position_measurement_interface =
+         std::make_shared<px4_ros2::GlobalPositionMeasurementInterface>(*this);
+   }
+
+   void sendUpdate()
+   {
+      while (running) { // Potentially make method run as a callback or on a timer
+         // Generate global position measurement
+         rclcpp::Time timestamp_sample  = ...
+         Eigen::Vector2d lat_lon = ...
+         float horizontal_variance = ...
+         float altitude_msl = ...
+         float vertical_variance = ...
+
+         // Populate the global position measurement struct [2]
+         px4_ros2::GlobalPositionMeasurement global_position_measurement{};
+         global_position_measurement.timestamp_sample = timestamp_sample;
+         global_position_measurement.lat_lon = lat_lon;
+         global_position_measurement.horizontal_variance = horizontal_variance;
+         global_position_measurement.altitude_msl = altitude_msl;
+         global_position_measurement.vertical_variance = vertical_variance;
+
+         // Send measurement to PX4 using the interface [3]
+         try {
+            _global_position_measurement_interface->update(global_position_measurement);
+         } catch (const px4_ros2::NavigationInterfaceInvalidArgument & e) {
+            // Handle exceptions caused by invalid global_position_measurement definition
+            RCLCPP_ERROR(get_logger(), "Exception caught: %s", e.what());
+         }
+      }
+   }
+
+private:
+   std::shared_ptr<px4_ros2::GlobalPositionMeasurementInterface> _global_position_measurement_interface;
+};
+```
+
+## Multiple Instances of an Interface
```

Comment:
FYI I moved this note from the "How to use" intro to its own section. It didn't feel right as a note.